docs/how_pool_works.md

```diff
@@ -0,0 +1,189 @@
+This document is an introduction to understanding how the funding pool contract works.
+It doesn't describe all the implementation details or the API,
+it focuses on the basic principles of the core mechanics.
+
+# Overview
+
+The funding pool is a smart contract, which creates real-time streams of donations.
+One can start, alter or end the process of sending his funds at any time with immediate effect.
+The flow of funds is automatically maintained and steady over time.
+
+There are 3 roles present in the contract.
+Any Ethereum address can be used in any number of these roles at the same time.
+
+- The sender - has assets and chooses who does he want to send them to, how much and how fast
+- The receiver - receives funds from senders
+- The proxy - receives funds from senders, but immediately passes them to receivers of his choice
+
+## The cycles
+
+The whole blockchain history is divided into cycles of equal length.
+Every cycle contains the same, constant number of blocks.
+In the examples below, we assume that the cycle length is 5.
+
+![](how_pool_works_1.png)
+
+# The receiver
+
+The receiver requires no preparation, any address can receive donations at any time.
+The only function of this role is collection of funds sent by others.
+
+## The deltas
+
+On every block, a sender can send funds to a receiver.
+The receiver can collect funds sent on a given block only when the cycle containing it is over.
+
+![](how_pool_works_2.png)
+
+Here we have a history of a receiver, who has 2 senders.
+Each of them was sending different amounts over different periods of time.
+On the end of each cycle, the collectable amount was increased by the total sent amount.
+
+Each receiver needs to know, how much was sent to him.
+To accomplish that the history of received funds needs to be stored in some form.
+The receiver doesn't care, who exactly is sending on each block and in what configuration.
+The only thing that matters are the collectable per-cycle values below the timeline.
+This reduces the amount of stored data significantly, but this can be optimized further.
+
+![](how_pool_works_3.png)
+
+In this example, we started with having the raw collectable value of 23 for every block until
+the end of the blockchain.
```

Comment:
what do you mean here "until the end of the blockchain?"

Reply:
I mean for infinite number of blocks higher than this one, like in "until the end of time". The blockchain obviously has no end :D

Reply:
"End of time" might be clearer because to me, the end of the chain is the head/tip :stuck_out_tongue:

Reply:
Fixed